tests/integration/api_asset_test.py

```diff
@@ -244,3 +269,467 @@ def test_restore_path(
     assert restored.guid == api_path.guid
     assert restored.qualified_name == api_path.qualified_name
     assert restored.status == EntityStatus.ACTIVE
+
+
+@pytest.fixture(scope="module")
+def api_object(
+    client: AtlanClient, connection: Connection
+) -> Generator[APIObject, None, None]:
+    assert connection.qualified_name
+    to_create = APIObject.create(
+        name=API_OBJECT_NAME, connection_qualified_name=connection.qualified_name
+    )
+    response = client.asset.save(to_create)
+    result = response.assets_created(asset_type=APIObject)[0]
+    yield result
+    delete_asset(client, guid=result.guid, asset_type=APIObject)
+
+
+def test_api_object(client: AtlanClient, connection: Connection, api_object: APIObject):
+    assert api_object
+    assert api_object.guid
+    assert api_object.qualified_name == API_OBJECT_QUALIFIED_NAME
+    assert api_object.name == API_OBJECT_NAME
+    assert api_object.connection_qualified_name == connection.qualified_name
+    assert api_object.connector_name == AtlanConnectorType.API.value
+
+
+@pytest.fixture(scope="module")
+def api_object_overload(
+    client: AtlanClient, connection: Connection
+) -> Generator[APIObject, None, None]:
+    assert connection.qualified_name
+    to_create = APIObject.creator(
+        name=API_OBJECT_NAME,
+        connection_qualified_name=connection.qualified_name,
+        api_field_count=API_OBJECT_FIELD_COUNT,
+    )
+    response = client.asset.save(to_create)
+    result = response.assets_created(asset_type=APIObject)[0]
+    yield result
+    delete_asset(client, guid=result.guid, asset_type=APIObject)
+
+
+def test_api_object_overload(
+    client: AtlanClient, connection: Connection, api_object_overload: APIObject
+):
+    assert api_object_overload
+    assert api_object_overload.guid
+    assert api_object_overload.qualified_name == API_OBJECT_QUALIFIED_NAME
+    assert api_object_overload.name == API_OBJECT_NAME
+    assert api_object_overload.connection_qualified_name == connection.qualified_name
+    assert api_object_overload.api_field_count == API_OBJECT_FIELD_COUNT
+    assert api_object_overload.connector_name == AtlanConnectorType.API.value
+
+
+@pytest.fixture(scope="module")
+def api_query(
+    client: AtlanClient, connection: Connection
+) -> Generator[APIQuery, None, None]:
+    assert connection.qualified_name
+    to_create = APIQuery.create(
+        name=API_QUERY_NAME, connection_qualified_name=connection.qualified_name
+    )
+    response = client.asset.save(to_create)
+    result = response.assets_created(asset_type=APIQuery)[0]
+    yield result
+    delete_asset(client, guid=result.guid, asset_type=APIQuery)
```

Comment:
We should use `creator()` instead (ref: https://github.com/atlanhq/atlan-python/pull/403/files#r1800760792)